docs/en/sql-reference/data-types/float.md

```diff
@@ -6,15 +6,33 @@ sidebar_label: Float32, Float64
 
 # Float32, Float64
 
+:::warning
+If you need accurate calculations, in particular if you work with financial or business data requiring a high precision you should consider using decimal instead. Floats might lead to innacurate results as illustrated below:
+
+```
+CREATE TABLE IF NOT EXISTS float_vs_decimal
+(
+   my_float Float64,
+   my_decimal Decimal64(3)
+)Engine=MergeTree ORDER BY tuple()
+
+INSERT INTO float_vs_decimal SELECT round(canonicalRand(), 3) AS res, res FROM system.numbers LIMIT 1000000; # Generate 1 000 000 random number with 2 decimal places and store them as a float and as a decimal
+
+SELECT sum(my_float), sum(my_decimal) FROM float_vs_decimal;
+> 500279.56300000014	500279.563
```

Comment:
I think we can make this example better.

From wiki

> The 11 bit width of the exponent allows the representation of numbers between 10−308 and 10308, with full 15–17 decimal digits precision. By compromising precision, the subnormal representation allows even smaller values up to about 5 × 10−324. 


012345 67890123456
500279.56300000014

So, it's clearly visible, that our number is correct up to ~15 decimal digits precision, so it's working according to standard. 

And `sumKahan` [Kahan](https://en.wikipedia.org/wiki/Kahan_summation_algorithm) is used mostly to solve another issue with Float64 calculation:

```
SELECT (toFloat64(100000000000000000.) + toFloat64(7.5)) - toFloat64(100000000000000000.) AS res

┌─res─┐
│   0 │
└─────┘


SELECT (toFloat64(100000000000000000.) - toFloat64(100000000000000000.)) + toFloat64(7.5) AS res

┌─res─┐
│ 7.5 │
└─────┘
```

> With compensated summation, using a compensation variable with sufficiently high [precision](https://en.wikipedia.org/wiki/Precision_(arithmetic)) the worst-case error bound is effectively independent of n n, so a large number of values can be summed with an error that only depends on the floating-point precision of the result. (It's already working in ~ that way in this example)